### PR TITLE
allow non-integer fhrot

### PR DESCRIPTION
**Pull request by junwang-noaa**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Require specific package version in top-level CMakeLists.txt.
Rename EARTH_GRID_COMP to UFSDriver.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #1017
- fixes #1162
- fixes #1240

## Testing

How were these changes tested? regression test on Hera.
What compilers / HPCs was it tested with? Intel/Hera
Are the changes covered by regression tests? Yes.
Have regression tests and unit tests (utests) been run? Not yet.
On which platforms and with which compilers? Hera/Intel

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

No.

### CODE REVIEW

File: driver/UFS.F90 [hunk 1]
```diff
@@ -347,16 +348,15 @@ PROGRAM UFS
 !***  This will correctly set the UFS Driver clocks in case of
 !***  Restart-From-History.
 !-----------------------------------------------------------------------
-
+      fhrot = 0.
       CALL ESMF_ConfigGetAttribute(config   = CF_MAIN  &
                                    ,value   = fhrot    &
                                    ,label   = 'fhrot:' &
-                                   ,default = 0        &
```
Comment (DusanJovic-NOAA) [inline driver/UFS.F90:354]: Why is default removed? And replaced with explicit initialization of fhrot.

Comment (junwang-noaa) [reply driver/UFS.F90:354]: It gives me compile error when setting to (0. ). I checked the ESMF manual, https://earthsystemmodeling.org/docs/release/ESMF_8_1_1/ESMF_refdoc.pdf, unfortunately it takes character string!

Comment (DusanJovic-NOAA) [reply driver/UFS.F90:354]: fhrot is now real (i.e. double precision), so 0.0 should be valid default value. try to declare fhrot as:
real(ESMF_KIND_R8) :: fhrot

Comment (junwang-noaa) [reply driver/UFS.F90:354]: That was my first try.  It did not work.

Comment (DusanJovic-NOAA) [reply driver/UFS.F90:354]: Both `fhrot` and literal `0.0` in default should be ESMF_KIND_R8, so:

```
      real(ESMF_KIND_R8) :: fhrot

      . . .

      CALL ESMF_ConfigGetAttribute(config   = CF_MAIN          &                                                       
                                   ,value   = fhrot            &                                                       
                                   ,label   = 'fhrot:'         &                                                       
                                   ,default = 0.0_ESMF_KIND_R8 &                                                       
                                   ,rc      = RC)  
```

Comment (junwang-noaa) [reply driver/UFS.F90:354]: Thanks, that works.

**Pull request merged**